src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,177 @@
+    /// @param state Initial state of the order
+    event RedemptionOrderCreated(
+        uint indexed orderId,
+        address indexed seller,
+        uint sellAmount,
+        uint exchangeRate,
+        uint collateralAmount,
+        uint feePercentage,
+        uint feeAmount,
+        uint redemptionAmount,
+        address collateralToken,
+        uint redemptionTime,
+        RedemptionState state
+    );
+
+    /// @notice Emitted when a redemption order's state changes
+    /// @param orderId Order identifier
+    /// @param newState New state of the order
+    event RedemptionOrderStateUpdated(
+        uint indexed orderId, RedemptionState newState
+    );
+
+    //--------------------------------------------------------------------------
+    // Errors
+
+    /// @notice Thrown when batch operation exceeds size limit
+    /// @param size_ Size of the batch attempted
+    error FM_ExternalPrice__BatchSizeExceeded(uint size_);
+
+    /// @notice Thrown when attempting to interact with zero address
+    error FM_ExternalPrice__ZeroAddress();
+
+    /// @notice Thrown when an invalid amount is provided
+    error FM_ExternalPrice__InvalidAmount();
+
+    /// @notice Thrown when interacting with a blacklisted address
+    /// @param account_ The blacklisted address
+    error FM_ExternalPrice__AddressBlacklisted(address account_);
+
+    /// @notice Thrown when non-whitelisted address attempts restricted operation
+    /// @param account_ The non-whitelisted address
+    error FM_ExternalPrice__NotWhitelisted(address account_);
+
+    /// @notice Thrown when an invalid address is provided
+    error FM_ExternalPrice__InvalidAddress();
+
+    /// @notice Fee exceeds maximum allowed value
+    /// @param fee The fee that was attempted to be set
+    /// @param maxFee The maximum allowed fee
+    error FM_ExternalPrice__FeeExceedsMaximum(uint256 fee, uint256 maxFee);
+
+    //--------------------------------------------------------------------------
+    // View Functions
+
+    /// @notice Gets the maximum number of addresses that can be whitelisted in a batch
+    /// @return maxBatchSize_ The maximum batch size
+    function getMaxBatchSize() external view returns (uint maxBatchSize_);
+
+    /// @notice Gets the current open redemption amount
+    /// @return amount_ The total amount of open redemptions
+    function getOpenRedemptionAmount() external view returns (uint amount_);
+
+    /// @notice Gets the next available order ID
+    /// @return orderId_ The next order ID
+    function getNextOrderId() external view returns (uint orderId_);
+
+    /// @notice Gets the current order ID
+    /// @return orderId_ The current order ID
+    function getOrderId() external view returns (uint orderId_);
+
+    /// @notice Buys tokens with provided collateral
+    /// @param collateralAmount_ The amount of collateral to spend
+    function buy(uint collateralAmount_, uint minAmountOut_) external;
```

Comment:
I don't think we need to define the buy and sell function in the interface as they are defined in the base contracts

Reply:
removed from interface